nifi-framework-api/src/main/java/org/apache/nifi/controller/queue/FlowFileQueue.java

```diff
@@ -93,6 +93,18 @@
 
     QueueSize size();
 
+    /**
+     * @param fromTimestamp The timestamp in milliseconds from which to calculate durations. This will typically be the current timestamp.
+     * @return the sum in milliseconds of how long all FlowFiles within this queue have currently been in this queue.
+     */
+    long getTotalActiveQueuedDuration(long fromTimestamp);
```

Comment:
I'm not really sure what exactly this metric is telling us. How would this metric be used? This would also become very expensive to calculate with data that is swapped out...

Reply:
This was an attempt to restrict the scope of this particular metric to flow files that were not swapped but in hindsight that isn't necessarily useful so will update the logic to include them as well per your suggestion. This metric is telling us the sum of the queued duration of all flow files within a given connection. My use case is to track both the average as well as the max duration to find bottlenecks/issues. I chose to report the total vs calculating and reporting an average to match the logic of other metrics contained within the ConnectionStatus class (InputBytes and OutputBytes).